src/script/view_model/ActionsViewModel.ts

```diff
@@ -29,28 +29,38 @@ import {PrimaryModal, removeCurrentModal, usePrimaryModalState} from 'Components
 import {t} from 'Util/LocalizerUtil';
 import {isBackendError} from 'Util/TypePredicateUtil';
 
+import type {MainViewModel} from './MainViewModel';
+
 import type {ClientEntity} from '../client';
 import type {ConnectionRepository} from '../connection/ConnectionRepository';
 import type {ConversationRepository} from '../conversation/ConversationRepository';
+import {ConversationState} from '../conversation/ConversationState';
 import type {MessageRepository} from '../conversation/MessageRepository';
 import {NOTIFICATION_STATE} from '../conversation/NotificationSetting';
 import type {Conversation} from '../entity/Conversation';
 import type {Message} from '../entity/message/Message';
 import type {User} from '../entity/User';
 import type {IntegrationRepository} from '../integration/IntegrationRepository';
 import type {ServiceEntity} from '../integration/ServiceEntity';
+import {ListState} from '../page/useAppState';
 import {SelfRepository} from '../self/SelfRepository';
 import {UserState} from '../user/UserState';
 
 export class ActionsViewModel {
+  public readonly mainViewModel: MainViewModel;
+  private readonly conversationState: ConversationState;
   constructor(
     private readonly selfRepository: SelfRepository,
     private readonly connectionRepository: ConnectionRepository,
     private readonly conversationRepository: ConversationRepository,
     private readonly integrationRepository: IntegrationRepository,
     private readonly messageRepository: MessageRepository,
     private readonly userState = container.resolve(UserState),
-  ) {}
+    mainViewModel: MainViewModel,
+  ) {
+    this.mainViewModel = mainViewModel;
+    this.conversationState = container.resolve(ConversationState);
```

Comment:
suggestion: Please put those 2 as `private readonly` in the params of the constructor